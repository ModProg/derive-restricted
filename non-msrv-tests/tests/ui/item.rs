use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where]
struct NoOption<T>(PhantomData<T>);

#[derive_where()]
struct EmptyAttribute<T>(PhantomData<T>);

#[derive_where(crate(derive_where_))]
struct WrongCrateSyntax<T>(PhantomData<T>);

#[derive_where(crate = "struct Test")]
struct InvalidPath<T>(PhantomData<T>);

#[derive_where(crate = "derive_where_", crate = "derive_where_")]
struct DuplicateCrate1<T>(PhantomData<T>);

#[derive_where(crate = "derive_where_")]
struct OnlyCrate<T>(PhantomData<T>);

#[derive_where(crate = "::derive_where")]
struct DefaultCrate<T>(PhantomData<T>);

#[derive_where(Clone; T;)]
struct SemiColonAtTheEnd<T, U>(T, PhantomData<U>);

#[derive_where(Clone; T,,)]
struct DoubleColonAtTheEnd<T, U>(T, PhantomData<U>);

#[derive_where(Clone; where)]
struct InvalidGeneric<T>(PhantomData<T>);

#[derive_where(Clone Debug)]
struct MissingCommaBetweenTraits<T>(PhantomData<T>);

#[derive_where(Clone; T U)]
struct MissingCommaBetweenGenerics<T, U, V>(T, PhantomData<(U, V)>);

#[derive_where("Clone")]
struct InvalidTrait<T>(PhantomData<T>);

fn main() {}
