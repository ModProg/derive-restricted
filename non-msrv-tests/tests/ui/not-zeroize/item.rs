use std::marker::PhantomData;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(skip_inner, Clone)]
struct SkipInnerWithTrait<T>(PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Debug = invalid; T)]
struct WrongOptionSyntax<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(,)]
struct OnlyComma<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(,Clone)]
struct StartWithComma<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Clone,,)]
struct DuplicateCommaAtEnd<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(T)]
struct GenericInsteadTrait<T, U>(T, PhantomData<U>);

fn main() {}
