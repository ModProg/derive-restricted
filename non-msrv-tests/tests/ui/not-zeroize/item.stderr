error: unsupported trait, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
 --> tests/ui/not-zeroize/item.rs:6:16
  |
6 | #[derive_where(skip_inner, Clone)]
  |                ^^^^^^^^^^

error: unsupported trait syntax, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:10:24
   |
10 | #[derive_where(Debug = invalid; T)]
   |                        ^^^^^^^

error: unsupported trait syntax, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:14:16
   |
14 | #[derive_where(,)]
   |                ^

error: unsupported trait syntax, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:18:16
   |
18 | #[derive_where(,Clone)]
   |                ^

error: unsupported trait syntax, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:22:22
   |
22 | #[derive_where(Clone,,)]
   |                      ^

error: unsupported trait, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
  --> tests/ui/not-zeroize/item.rs:26:16
   |
26 | #[derive_where(T)]
   |                ^
